lib/oli/rendering/activity/html.ex

```diff
@@ -109,23 +109,45 @@ defmodule Oli.Rendering.Activity.Html do
 
           attempts ->
             {:safe, attempt_selector} =
-              ReactPhoenix.ClientSide.react_component("Components.AttemptSelector", %{
-                activityId: activity_id,
-                attempts:
-                  Enum.map(attempts, fn a ->
-                    %{
-                      state: a.lifecycle_state,
-                      attemptNumber: a.attempt_number,
-                      attemptGuid: a.attempt_guid,
-                      date:
-                        Timex.format!(
-                          a.updated_at,
-                          "{Mfull} {D}, {YYYY} at {h12}:{m} {AM} {Zabbr}"
-                        )
-                    }
-                  end),
-                sectionSlug: section_slug
-              })
+              case liveview do
+                true ->
+                  PhoenixLiveReact.live_react_component("Components.AttemptSelector", %{
```

Comment:
In an effort to eliminate duplicated ReactPhoenix/ReactLivePhoenix react component definitions like these, I created a module `OliWeb.Common.React` `lib/oli_web/common/react.ex` to handle both cases. Though currently, it only accepts a SessionContext struct I think we could expand it to just accept an `is_liveview` param as well, then we could use this single common component here instead of defining twice? What do you think?

```
defmodule OliWeb.Common.React do
  # use Phoenix.Component

  import PhoenixLiveReact

  alias OliWeb.Common.SessionContext

  def component(_ctx_or_is_liveview?, name, props, opts \\ %{})

  def component(%SessionContext{is_liveview: true}, name, props, opts),
    do: live_react_component(name, props, opts)

  def component(%SessionContext{is_liveview: false}, name, props, opts),
    do: ReactPhoenix.ClientSide.react_component(name, props, opts)

  def component(true = _is_liveview, name, props, opts),
    do: live_react_component(name, props, opts)

  def component(false = _is_liveview, name, props, opts),
    do: ReactPhoenix.ClientSide.react_component(name, props, opts)
end

```

Reply:
Yup, that sounds great! Maybe we can change it so that it accepts a map instead of a `SessionContext`? WDYT?